src/pages/home/sidebar/ChatSwitcherView.js

```diff
@@ -461,12 +466,47 @@ class ChatSwitcherView extends React.Component {
             }
         }
 
-        this.setState({
-            options: Array.from(matches),
-        });
+        let options = Array.from(matches);
+        if (options.length === 0) {
+            const searchStr = value;
+            if (searchStr && searchStr.includes('@') && Str.isValidEmail(searchStr)) {
+                options = [{
+                    text: searchStr,
+                    alternateText: searchStr,
+                    singleUserDM: true,
+                    type: CONST.OPTION_TYPE.PERSONAL_DETAIL,
+                    keyForList: searchStr,
+                    login: searchStr,
+                }];
+            }
+
+            const isPhoneNumber = /^[+|\d]+$/.test(searchStr);
+            if (searchStr && isPhoneNumber && Str.isValidPhone(searchStr)) {
+                const phoneNumber = searchStr.includes('+') ? searchStr : `+${this.props.countryCodeByIP}${searchStr}`;
```

Comment:
I'm a little confused by all of this logic and finding it hard to tell if we are missing something. Just going to rubber duck you tell me if it sounds right (and then maybe we can add some comments).

- If the search string has a `+` we assume the user knows their country code and has included it
- Otherwise we are going to add the `+` and the country code
- But then the `login` (which I think we create the chat from) will either have the country code or not?

I think this will mean that if I type my number like this

`15165677123`

it will show up as

`+115165677123`

Which might not be expected... just wondering if there is anything more we can do here. But if not, we should explain this maybe so it's understood.

Reply:
> If the search string has a + we assume the user knows their country code and has included it
Otherwise we are going to add the + and the country code
But then the login (which I think we create the chat from) will either have the country code or not?

All correct. 

> I think this will mean that if I type my number... Which might not be expected.

That's what tom asked me if we could add so i added it here. Sure i can add a comment above explaining this.